### ISSUE TITLE
bad output formatting on windows command prompt

### ISSUE THREAD

**Issue by sylvainpolletvillard** [lukeed/taskr#101]
![image](https://cloud.githubusercontent.com/assets/566536/9836531/a9ecf484-5a1d-11e5-9c8a-3fb68558f173.png)

Fly outputs have weird formatting on Windows (Windows 10 x64 for me).

Someone has suggested it could come from ANSI Color codes. I tried many things including alternative sheels, ansicon and console emulators yet none of these worked.

I also struggled to properly add fly to my environment variables on Windows without installing fly globally, but that is another issue. Anyway, there is still work to do to use Fly on Windows comfortably.

**Comment by ghost** [comment]
This is an epic fail, but I know people have got Fly working on Windows alright.

> Anyway, there is still work to do to use Fly on Windows comfortably.

So it seems. I don't use Windows, so it's rather difficult to test this comfortably for me too, but I will definitely get around this before 1.0 sometime this month.

**Comment by sylvainpolletvillard** [comment]
Well, Fly is working for me on Windows at the moment, but there are stille some details like this one to fix

**Comment by gaokun** [comment]
@sylvainpolletvillard  Hi, this issue happened to me, :(.     U said it works for u. Did u do something to fix that? Thx in advance.

**Comment by sylvainpolletvillard** [comment]
@gaokun: Yes, I used this : https://github.com/adoxa/ansicon ; but there were other problems beside formatting so maybe it's wiser to just use cygwin.

**Comment by gaokun** [comment]
@sylvainpolletvillard  Thanks man, never heard cygwin, that helps a lot. ^_^

**Comment by hzlmn** [comment]
@lukeed @bucaran Is it still an issue? Sadly, I have Windows installed in parallel, so I can look on this.

**Comment by lukeed** [comment]
@hzlmn No idea 😟 Havent touched windows in years

**Comment by hzlmn** [comment]
@lukeed Lucky man 😄

**Comment by lukeed** [comment]
Related? http://superuser.com/questions/269818/change-default-code-page-of-windows-console-to-utf-8

**Comment by lukeed** [comment]
@devmondo have you or your WIndows team ever run into this? Not sure if it's still applicable

**Comment by devmondo** [comment]
@lukeed  we use webstorm terminal,Windows CMD and ConEmu and never faced this issue, the only issue facing is that even though we install Fly globally we cant run it by typing `fly in CMD we have to do`npm run fly`.

using Windows 10 x64

**Comment by lukeed** [comment]
@devmondo thanks! If you could run a couple builds within a standard `cmd.exe` & let me know how that goes, it'd be much appreciated :) It sounds like of all the currently active participants, you're the only one who is on Windows natively (not a VM)... and the few times I have been, I never ran into this problem. 

Cheers! 🍻

**Comment by devmondo** [comment]
@lukeed i ran more than couple of builds with cmd.exe, Fly replaced all our Gulp cr*ap now, and it is working solid after the `file` bug :clap: and none of weird characters mentioned above show, only problem as i mentioned is we cant run `fly` command like `gulp` it has to be `npm run`fly` as i mentioned, is this on purpose ?

and yes we are Windows shop so let me know if you need anymore tests. :neckbeard:

@sylvainpolletvillard maybe this helps? [fix CMD](https://github.com/npm/npm/issues/10648#issuecomment-163679125)

**Comment by lukeed** [comment]
Cool, thanks @devmondo! Great link too. 

Going to go ahead & close this due to (1) inactivity, (2) no longer applies, and/or (3) is not specific to Fly.

**Comment by sylvainpolletvillard** [comment]
@devmondo I will try again with latest Fly this evening and see if your solution helps

**Comment by sylvainpolletvillard** [comment]
Still have the issue with Fly v1.4.1 on both cmd.exe and Webstorm terminal on Windows 10
![image](https://cloud.githubusercontent.com/assets/566536/19125082/01fad372-8b36-11e6-8027-251a8b5118bc.png)

@devmondo your link points to a solution for MacOS, not for Windows. I did not find a similar config parameter on Windows.

**Comment by lukeed** [comment]
@sylvainpolletvillard Do you have any sort of plugins or settings running for your console? If so, can you try disabling them?

I have access to a Windows machine that has no customization whatsoever, and everything displays as it should.

In fact, your situation is so unique that I literally can't find any results when searching for vaguely related issues.

**Comment by sylvainpolletvillard** [comment]
Not that I'm aware of. Maybe it is related to my system language. I'll keep searching

**Comment by lukeed** [comment]
That might make sense 🤔

**Comment by devmondo** [comment]
@sylvainpolletvillard sorry for that, are you sure you have unicode enabled in CMD? did you try other terminals like conemu, hell even git bash?

**Comment by sylvainpolletvillard** [comment]
I tried changing the code page to UTF-8 and UTF-7, no success.

It works with ANSICON (https://github.com/adoxa/ansicon). This is probably related to ANSI escape sequences. I thought it could come from [clor ](https://github.com/jbucaran/clor) but the examples from this repo are working perfectly.

**Comment by sylvainpolletvillard** [comment]
Found it ! These '~' characters come from here : https://github.com/flyjs/fly/blob/master/lib/utils/logging.js#L77

process.env.HOME is undefined on my system

PR incoming

**Comment by lukeed** [comment]
Ohh interesting. I didn't know that!
